api/store/mongo/migrations.go

```diff
@@ -1,11 +1,39 @@
 package mongo
 
 import (
+	"context"
+
 	"github.com/shellhub-io/shellhub/api/store/mongo/migrations"
+	"github.com/sirupsen/logrus"
+	"github.com/square/mongo-lock"
 	migrate "github.com/xakep666/mongo-migrate"
 	"go.mongodb.org/mongo-driver/mongo"
 )
 
 func ApplyMigrations(db *mongo.Database) error {
-	return migrate.NewMigrate(db, migrations.GenerateMigrations()...).Up(migrate.AllAvailable)
+
+	logrus.Info("Creating lock for the resource migrations")
+
+	lockClient := lock.NewClient(db.Collection("migrations"))
+	if err := lockClient.CreateIndexes(context.TODO()); err != nil {
+		logrus.WithError(err).Fatal("Failed to create a lock for the database")
+	}
+
+	logrus.Info("Locking the resource migrations")
+
+	lockId := "0"
+
+	if err := lockClient.XLock(context.TODO(), "migrations", lockId, lock.LockDetails{}); err != nil {
```

Comment:
Do we need to use write consensus here?

Reply:
Yes, because every process has her own instance of the store. But, they share the same resource which is the database, so we need to establish a write consensus.